components/bl-dropzone-component/src/helpers.js

```diff
@@ -0,0 +1,97 @@
+import { useState } from 'react';
+
+const UploadStatus = {
+  success  : 'success',
+  uploading: 'uploading',
+  undefined: 'undefined',
+  error    : 'error',
+};
+
+export function useDropzone(component, eventHandlers) {
+  const { overwriteFiles, targetDirectory, uploadOnDrop } = component;
+  const { onDelete, onChange, fileNameLogic, onUpload, available } = eventHandlers;
+
+  const [files, setFiles] = useState([]);
+
+  const updateFiles = filesList => {
+    const updatedFiles = [];
+
+    filesList.forEach(async item => {
+      if (item.uploadStatus === UploadStatus.uploading) {
+        item.uploadStatus = UploadStatus.undefined;
+
+        try {
+          const file = item.file;
+          const hasUploadHandler = available('onUpload');
+
+          if (hasUploadHandler) {
+            await onUpload({ file });
+          } else {
+            const fileName = await fileNameLogic({ file });
+            const validFileName = ensureExtension(fileName, file);
+            const directory = targetDirectory?.replace(/\/$/g, '').replace(/(.*)/, '$1/') || '';
+            const path = directory + validFileName;
+
+            await Backendless.Files.upload(file, path, overwriteFiles);
+          }
+
+          item.uploadStatus = UploadStatus.success;
+        } catch (error) {
+          item.uploadStatus = UploadStatus.error;
+          console.error(error.message);
```

Comment:
would be better to remember the error and pass to a Codeless Logic instead of printing it here